Anlab.Core/Services/AggieEnterpriseService.cs

```diff
@@ -0,0 +1,208 @@
+using AggieEnterpriseApi;
+using AggieEnterpriseApi.Extensions;
+using AggieEnterpriseApi.Types;
+using AggieEnterpriseApi.Validation;
+using Anlab.Core.Models.AggieEnterpriseModels;
+using Microsoft.Extensions.Options;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Anlab.Core.Services
+{
+    public interface IAggieEnterpriseService
+    {
+        Task<AccountValidationModel> IsAccountValid(string financialSegmentString, bool validateCVRs = true);
+
+        Task<FinancialOfficerDetails> GetFinancialOfficer(string financialSegmentString);
+     
+        Task<string> ConvertKfsAccount(string account);
+    }
+
+    public class AggieEnterpriseService : IAggieEnterpriseService
+    {
+        private readonly IAggieEnterpriseClient _aggieClient;
+
+        public AggieEnterpriseSettings AeSettings { get; set; }
+
+        public AggieEnterpriseService(IOptions<AggieEnterpriseSettings> aggieEnterpriseSettings)
+        {
+            AeSettings = aggieEnterpriseSettings.Value;
+
+            try
+            {
+                _aggieClient = GraphQlClient.Get(AeSettings.GraphQlUrl, AeSettings.Token);
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error creating Aggie Enterprise Client");
+                _aggieClient = null;
+            }
+        }
+
+        public async Task<AccountValidationModel> IsAccountValid(string financialSegmentString, bool validateCVRs = true)
+        {
+            var rtValue = new AccountValidationModel();
+            rtValue.IsValid = false;
+            rtValue.FinancialSegmentString = financialSegmentString;
+
+            var segmentStringType = FinancialChartValidation.GetFinancialChartStringType(financialSegmentString);
+            rtValue.CoaChartType = segmentStringType;
+
+            if (segmentStringType == FinancialChartStringType.Gl)
+            {
+                var result = await _aggieClient.GlValidateChartstring.ExecuteAsync(financialSegmentString, validateCVRs);
+
+                var data = result.ReadData();
+
+                rtValue.IsValid = data.GlValidateChartstring.ValidationResponse.Valid;
+
+                if (!rtValue.IsValid)
+                {
+                    foreach (var err in data.GlValidateChartstring.ValidationResponse.ErrorMessages)
+                    {
+                        rtValue.Messages.Add(err);
+                    }
+                }
+                rtValue.Details.Add(new KeyValuePair<string, string>("Entity", $"{data.GlValidateChartstring.SegmentNames.EntityName} ({data.GlValidateChartstring.Segments.Entity})"));

```

Comment:
I probably don't need these details, but going to leave in there for now in case we want to display any of this info on the lab details page.